src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1287,6 +1287,19 @@ void IVisualStudioWorkspaceHost.OnAdditionalDocumentTextUpdatedOnDisk(DocumentId
                 _workspace.OnAdditionalDocumentTextUpdatedOnDisk(id);
             }
 
+            void IVisualStudioWorkspaceHost.OnIntellisenseBuildResult(ProjectId projectId, bool succeed)
+            {
+                // this can't be null
+                var state = _workspace.CurrentSolution.GetProjectState(projectId);
+                Contract.ThrowIfNull(state);
+
+                // if state is different than what we have
+                if (succeed != state.ProjectInfo.IsComplete)
+                {
+                    _workspace.OnProjectReloaded(state.ProjectInfo.WithIsComplete(succeed));
```

Comment:
we need to change projectInfo anyway, so this was simpler and most of time, it doesn't get called. instead nature ProjectInfo.Create in AbstractProject is used. when it actually get called, it is right after project load, so probably no cache is created yet.